File: example/java/jimmer-sql/src/main/java/org/babyfish/jimmer/sql/example/model/Chapter.java
package org.babyfish.jimmer.sql.example.model;

import org.babyfish.jimmer.pojo.AutoScalarStrategy;
import org.babyfish.jimmer.pojo.StaticType;
import org.babyfish.jimmer.sql.*;

@Entity
@StaticType(alias = "forCompositeBookInput", autoScalarStrategy = AutoScalarStrategy.DECLARED)
public interface Chapter {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id();

    @ManyToOne
    @Key
    @OnDissociate(DissociateAction.DELETE)
    Book book();

    @Key
    @Column(name = "chapter_no")
    int index();

    String title();
}

File: example/kotlin/jimmer-sql-kt/src/main/kotlin/org/babyfish/jimmer/example/kt/sql/model/Chapter.kt
package org.babyfish.jimmer.example.kt.sql.model

import org.babyfish.jimmer.example.kt.sql.model.common.BaseEntity
import org.babyfish.jimmer.pojo.AutoScalarStrategy
import org.babyfish.jimmer.pojo.StaticType
import org.babyfish.jimmer.sql.*

@Entity
@StaticType(alias = "declaredOnly", autoScalarStrategy = AutoScalarStrategy.DECLARED)
interface Chapter : BaseEntity {

    @Id
    val id: Long

    @Key
    @ManyToOne
    @OnDissociate(DissociateAction.DELETE)
    val book: Book

    @Key
    @Column(name = "chapter_no")
    val index: Int

    val title: String
}